### PR TITLE
feat(progressbar.js): new type definition

### PR DESCRIPTION
**Pull request by peterblazejewicz**
- definitioin files
- tests

https://github.com/kimmobrunfeldt/progressbar.js#progressbarjs
https://www.npmjs.com/package/progressbar.js

x-ref:
kimmobrunfeldt/progressbar.js#206
kimmobrunfeldt/progressbar.js#149

Thanks!

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` should be present and it shouldn't have any additional or disabling of rules.
> the Shape/Path contain exception to merge animation members into base classses
- [x] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/progressbar.js/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,209 @@
+// Type definitions for progressbar.js 1.1
+// Project: https://kimmobrunfeldt.github.io/progressbar.js/
+// Definitions by: Piotr Błażejewicz <https://github.com/peterblazejewicz>
```
Comment (danger-public) [inline types/progressbar.js/index.d.ts:3]: - :no_entry_sign: This line should have your github username in it, not /me

File: types/progressbar.js/progressbar.js-tests.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
+import ProgressBar = require('progressbar.js');
+import utils = require('progressbar.js/utils');
```
Comment (sheetalkamat) [inline types/progressbar.js/progressbar.js-tests.ts:2]: this is incorrect import..
```js
const utils = require('progressbar.js/utils');
```

results in 
```
c:\temp\test2>node a.js
internal/modules/cjs/loader.js:796
    throw err;
    ^

Error: Cannot find module 'progressbar.js/utils'
Require stack:
- c:\temp\test2\a.js
←[90m    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:793:17)←[39m
←[90m    at Function.Module._load (internal/modules/cjs/loader.js:686:27)←[39m
←[90m    at Module.require (internal/modules/cjs/loader.js:848:19)←[39m
←[90m    at require (internal/modules/cjs/helpers.js:74:18)←[39m
    at Object.<anonymous> (c:\temp\test2\a.js:2:15)
←[90m    at Module._compile (internal/modules/cjs/loader.js:955:30)←[39m
←[90m    at Object.Module._extensions..js (internal/modules/cjs/loader.js:991:10)←[39m
←[90m    at Module.load (internal/modules/cjs/loader.js:811:32)←[39m
←[90m    at Function.Module._load (internal/modules/cjs/loader.js:723:14)←[39m
←[90m    at Function.Module.runMain (internal/modules/cjs/loader.js:1043:10)←[39m {
  code: ←[32m'MODULE_NOT_FOUND'←[39m,
  requireStack: [ ←[32m'c:\\temp\\test2\\a.js'←[39m ]
}
```

**Review by sheetalkamat — CHANGES_REQUESTED** [review]

File: types/progressbar.js/progressbar.js-tests.ts (hunk 2 above)
Comment (peterblazejewicz) [reply types/progressbar.js/progressbar.js-tests.ts:2]: let me check, thanks!

Comment (peterblazejewicz) [reply types/progressbar.js/progressbar.js-tests.ts]: @sheetalkamat I amended with proper change (now I've just do the homework adding this into real progressbar.js code). Thanks for review!

**Pull request merged**